pipelineRunner.go

```diff
@@ -313,43 +310,47 @@ func (pr *pipelineRunnerImpl) RunStages(ctx context.Context, depth int, stages [
 	log.Info().Msgf("Running %v stages", len(stages))
 
 	var finalErr error
-	for _, p := range stages {
-
-		// handle cancellation happening in between stages
-		if pr.canceled {
-			// set canceled status for all the next stages
-			pr.forceStatusForStage(*p, contracts.StatusCanceled)
-			continue
-		}
-
-		var whenEvaluationResult bool
-		whenEvaluationResult, err = pr.whenEvaluator.Evaluate(p.Name, p.When, pr.whenEvaluator.GetParameters())
-		if err != nil {
-			return buildLogSteps, err
-		}
-
-		if whenEvaluationResult {
-
-			err = pr.RunStageWithRetry(ctx, depth, dir, envvars, nil, *p)
+	for _, s := range stages {
+		func(stage *manifest.EstafetteStage) {
```

Comment:
By wrapping it in a function I can use defer to make sure on cancellation the right status is set, otherwise the pipeline will wait forever with tailing logs.